lib/browser/api/menu.js

```diff
@@ -45,30 +45,49 @@ Menu.prototype._init = function () {
 
 Menu.prototype.popup = function (window, x, y, positioningItem) {
   let [newX, newY, newPosition, newWindow] = [x, y, positioningItem, window]
+  let opts
 
   // menu.popup(x, y, positioningItem)
-  if (!window) {
-    // shift over values
-    if (typeof window !== 'object' || window.constructor !== BrowserWindow) {
-      [newPosition, newY, newX, newWindow] = [y, x, window, null]
-    }
+  if (window && typeof window !== 'object' || window.constructor !== BrowserWindow) {
```

Comment:
Can this check be replaced with a simpler `if (window && !(window instance of BrowserWindow))`?

Reply:
Yeah I'm wondering what prompted that `typeof window !== 'object'` clause in the first place